src/main/java/com/google/devtools/build/lib/analysis/actions/CustomCommandLine.java

```diff
@@ -1128,14 +1115,7 @@ public NestedSet<Artifact> getTreeArtifactInputs() {
     }
 
     public CustomCommandLine build() {
-      return stripOutputPaths
```

Comment:
Re your description + this specific change:
> PathMappers are no longer passed into builders for structured command lines, but are supplied when expanding command lines. In addition to not retaining them, this also reduces the number of places in which participating actions have to pass in PathMappers.

I'm reading it as when `SpawnAction` is created the commandLines' arguments would remain unstripped and we no longer need PathStrippingCustomCommandline as these will be taken care of later stage during `CommandLines.expand`?

If that is possible, this change would be very beneficial for another change I'm working on to disable path stripping for local actions since it's not possible to get the resolved strategy at this layer. 

P.S: I still owe you a more detailed reviews of this pr and will get to it in the next few days.